scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -127,7 +127,7 @@ abstract class ScanResultsStorage {
     /**
      * The name to refer to this storage implementation.
      */
-    val name: String = javaClass.simpleName
+    open val name: String = javaClass.simpleName
```

Comment:
How about making the variable a constructor parameter (with a default value) instead?

Reply:
What's the benefit?

Reply:
I gave it a try to see how it looks like, but I ran into problems accessing `javaClass.simpleName` as a (default) constructor argument:

    Warning:(34, 38) Kotlin: The resulting type of this 'javaClass' call is Class<ScanResultsStorage.Companion> and not Class<ScanResultsStorage>. Please use the more clear '::class.java' syntax to avoid confusion

And I couldn't find a syntax that both compiles and does the right thing.

Reply:
Thought it was I tiny simplification but seems I was wrong.